plumbing/msg/sender.go

```diff
@@ -87,9 +107,17 @@ func (s *Sender) Send(ctx context.Context, from, to address.Address, value *type
 		return cid.Undef, errors.Wrap(err, "failed to marshal message")
 	}
 
-	// Add to the local message pool at the last possible moment before broadcasting to network.
-	if _, err := s.msgPool.Add(smsg); err != nil {
-		return cid.Undef, errors.Wrap(err, "failed to add message to the message pool")
+	height, err := s.blockTimer.BlockHeight()
+	if err != nil {
+		return cid.Undef, errors.Wrap(err, "failed to get block height")
+	}
+
+	// Add to the local message queue/pool at the last possible moment before broadcasting to network.
+	if err := s.outbox.Enqueue(smsg, height); err != nil {
```

Comment:
Non-Blocking: Still think block height should be the message queue's responsibility, not the callers. Calling Enqueue with anything other than the current block height will cause the timeout mechanism to misbehave.